### PR TITLE
[ROCM] Add ROCm support to debug_dump and enable_debug_mode

### PR DESCRIPTION
**Pull request by dllehr-amd**
enable_debug_mode and debug_dump are enabled in ROCM releases.  Add ROCM flags to #if defines so they can be accessed by PyTorch users.

Fixes #ISSUE_NUMBER

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang

### CODE REVIEW

File: aten/src/ATen/cuda/CUDAGraph.cpp [hunk 1]
```diff
@@ -243,7 +243,7 @@ void CUDAGraph::replay() {
 }
 
 void CUDAGraph::enable_debug_mode() {
-#if !defined(USE_ROCM) || ROCM_VERSION >= 50300
+#if (!defined(USE_ROCM) || ROCM_VERSION >= 50300) || (defined(USE_ROCM) && ROCM_VERSION >= 50600)
```
Comment (hongxiayang) [inline aten/src/ATen/cuda/CUDAGraph.cpp:246]: is (defined(USE_ROCM) && ROCM_VERSION >= 50600) necessary? as when if ROCM_VERSION >= 50300, this is already true.

Comment (dllehr-amd) [reply aten/src/ATen/cuda/CUDAGraph.cpp]: Don't code when tired :)  Took it off

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@huydhn @malfet ,
Can you please review this PR.

**Review by huydhn — APPROVED** [review]

**Pull request closed without merging**